test/utils/fixture/fixture.go

```diff
@@ -94,13 +96,63 @@ func SetupWithOptions(name string, nrtList nrtv1alpha2.NodeResourceTopologyList,
 	}
 	randomizeName := (options & OptionRandomizeName) == OptionRandomizeName
 	avoidCooldown := (options & OptionAvoidCooldown) == OptionAvoidCooldown
+	isSuiteSetup := (options & OptionIsSuite) == OptionIsSuite
+	ginkgo.By("set up the test namespace")
 	ns, err := setupNamespace(e2eclient.Client, name, randomizeName)
 	if err != nil {
 		klog.Errorf("cannot setup namespace %q: %v", name, err)
 		return nil, err
 	}
-	ginkgo.By(fmt.Sprintf("set up the test namespace %q", ns.Name))
+	klog.Infof("test namespace %q was set up successfully", ns.Name)
+	klog.Info("Going Once .... \n\n\n")
+	if !isSuiteSetup {
+		ginkgo.By("pull NRT data before test starts")
+		var nrtAtTestSetup nrtv1alpha2.NodeResourceTopologyList
+		immediate := true
+		err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 30*time.Second, immediate, func(ctx context.Context) (bool, error) {
+			err := e2eclient.Client.List(context.TODO(), &nrtAtTestSetup)
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
```

Comment:
good, better: 
```
err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 30*time.Second, immediate, func(ctx context.Context) (bool, error) {
	err := e2eclient.Client.List(context.Background(), &nrtAtTestSetup)
        return err == nil, nil
})
```

even better: create a single `context` at the beginning of this function and use it consistenly